packages/ra-ui-materialui/src/button/SaveButton.tsx

```diff
@@ -43,10 +45,10 @@ const SaveButton: FC<SaveButtonProps> = props => {
 
     const handleClick = event => {
         if (typeof onSave === 'function') {
-            setOnSave(onSave);
+            setOnSave && setOnSave(onSave);
```

Comment:
If the SaveButton is used outside of a FormContext, we want the save to fail.

Reply:
I was using type-script with strictMode = true. I'll change it.